### PR TITLE
ReportExport and VectorLink reports

### PR DESCRIPTION
**Pull request by lwyszomi**
Hi,

I changed the ```ReportExport``` class to use the data from the report which should be exported. We had a problem with the VectorLink report that UI reports work correctly but emailed reports not. This occurred because in ```ReportExport``` class we getting once again data source configuration from the database, I think we should use this from Report. In VectorLink reports, we using ```custom_configurable_report``` where in the class we override ```datasource``` function

Please let me know if you have any questions.

Regards,
Łukasz

### CODE REVIEW

File: custom/abt/reports/views.py [hunk 1]
```diff
@@ -244,10 +239,19 @@ def get_total_records(self):
...
+
+    @property
+    def config(self):
+        return self.original_data_source.config
+
 
 class FormattedSprayProgressReport(CustomConfigurableReport):
 
     @property
+    @memoized
```
[the remark this answers is not in the document]
Comment (lwyszomi) [reply custom/abt/reports/views.py]: @nickpell I wanted that my function looks the same as original:
https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/userreports/reports/view.py#L207

[the remark this answers is not in the document]
Comment (nickpell) [reply custom/abt/reports/views.py]: 👍

**Comment by esoergel** [comment]
Had to revert this - this change was sending an instantiated django view to celery as a parameter (`ReportExport(self)`), which couldn't be pickled.  Also, it looks like the export task depends on the `report_export.title` parameter being available.

> we getting once again data source configuration from the database

We generally try to avoid passing large objects as parameters to celery, so this is pretty standard practice.  Could you do this by passing simple parameters somehow?

**Comment by lwyszomi** [comment]
@esoergel I didn't notice that this is sending to celery, I will try to do something with this.

**Pull request merged**